### PR TITLE
Disallow `TypeVar` constraints parameterized by type variables

### PR DESCRIPTION
**Pull request by brianschubert**
Emit an error for [this test case](https://github.com/python/typing/blob/46b05a4c10ed3841c9bc5126ba9f31dd8ae061e7/conformance/tests/generics_basic.py#L54-L55) from the conformance tests:
```python
class Test(Generic[T]):
    BadConstraint2 = TypeVar("BadConstraint2", str, list[T])  # E
```

### CODE REVIEW

**Comment by brianschubert** [comment]
Conformance results diff for this PR (`generics_basic` now passes):
```diff
diff --git a/conformance/results/mypy/generics_basic.toml b/conformance/results/mypy/generics_basic.toml
index 00c8244..b15fe50 100644
--- a/conformance/results/mypy/generics_basic.toml
+++ b/conformance/results/mypy/generics_basic.toml
@@ -6,6 +6,7 @@ output = """
 generics_basic.py:40: error: Value of type variable "AnyStr" of "concat" cannot be "Sequence[object]"  [type-var]
 generics_basic.py:41: error: Value of type variable "AnyStr" of "concat" cannot be "Sequence[object]"  [type-var]
 generics_basic.py:49: error: Type variable must have at least two constrained types  [misc]
+generics_basic.py:55: error: TypeVar constraint type cannot be parametrized by type variables  [misc]
 generics_basic.py:69: error: Value of type variable "AnyStr" of "concat" cannot be "Sequence[object]"  [type-var]
 generics_basic.py:121: error: Duplicate type variables in Generic[...] or Protocol[...]  [misc]
 generics_basic.py:157: error: Invalid index type "int" for "MyMap1[str, int]"; expected type "str"  [index]
@@ -15,7 +16,6 @@ generics_basic.py:191: error: Type variable "generics_basic.T" is unbound  [vali
 generics_basic.py:191: note: (Hint: Use "Generic[T]" or "Protocol[T]" base class to bind "T" inside a class)
 generics_basic.py:191: note: (Hint: Use "T" in function signature to bind "T" inside a function)
 """
-conformance_automated = "Fail"
+conformance_automated = "Pass"
 errors_diff = """
-Line 55: Expected 1 errors
 """
diff --git a/conformance/results/mypy/version.toml b/conformance/results/mypy/version.toml
index 51d97b4..59e0dd2 100644
--- a/conformance/results/mypy/version.toml
+++ b/conformance/results/mypy/version.toml
@@ -1,2 +1,2 @@
-version = "mypy 1.14.0+dev.499adaed8adbded1a180e30d071438fef81779ec"
-test_duration = 2.4
+version = "mypy 1.14.0+dev.841096633fe6255acb7840e4a7b0fce975b8ca24"
+test_duration = 4.4
```

**Comment by brianschubert** [comment]
Two approaches, not sure which is better:
1. Original approach (https://github.com/python/mypy/pull/18186/commits/841096633fe6255acb7840e4a7b0fce975b8ca24), uses modified `FindTypeVarVisitor`, more complex, robust to placeholder types
2. Second approach ([`a8171c6`](https://github.com/python/mypy/pull/18186/files/a8171c6bebe13cbf99eaf989c69f86fa3af9f351)), uses `has_no_typevars`, simpler, but may have false negatives if a constraint type includes both a placeholder type and a type variable.

File: test-data/unit/semanal-errors.test [hunk 1]
```diff
@@ -1060,6 +1060,19 @@ S = TypeVar('S', covariant=True, contravariant=True) \
     # E: TypeVar cannot be both covariant and contravariant
 [builtins fixtures/bool.pyi]
 
+[case testInvalidTypevarArgumentsGenericConstraint]
```
Comment (JukkaL) [inline test-data/unit/semanal-errors.test:1063]: Can you also add a test case that uses the new type parameter syntax (in `test-data/unit/check-python312.test`)?

Comment (brianschubert) [reply test-data/unit/semanal-errors.test:1063]: Thanks, I missed the fact that PEP 695 type vars are processed in a different part. Added some handling and tests

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**